### PR TITLE
Fix compilation problems due to ed25519-dalek 1.0.0-pre.4

### PR DESCRIPTION
**Pull request by tomaka**
`ed25519-dalek` v1.0.0-pre.3 and v1.0.0-pre.4 are incompatible with each other. Unfortunately, Cargo doesn't seem to respect the semver rules here and treats them as compatible. The most painless way to fix this is to just upgrade to `v1.0.0-pre.4`.

polkadot companion: https://github.com/paritytech/polkadot/pull/1427

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -9281,7 +9298,7 @@ version = "1.5.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "3bfd5b7557925ce778ff9b9ef90e3ade34c524b5ff10e239c69a42d546d2af56"
 dependencies = [
- "rand 0.7.3",
+ "rand 0.3.23",
```
Comment (tomaka) [inline Cargo.lock:9301]: Err... that's another Cargo bug.

**Comment by tomaka** [comment]
Waiting for a fix for the companion status check:

> ./.maintain/gitlab/check_polkadot_companion_status.sh: line 90: syntax error: unexpected redirection

**Comment by DemiMarie-parity** [comment]
@tomaka you can fix this by fixing the syntax error.

**Comment by tomaka** [comment]
bot merge

**Pull request merged**